res/controllers/Pioneer-DDJ-400.midi.xml

```diff
@@ -1605,6 +1605,169 @@
             </control>
             <!-- BEAT LOOP MODE END -->
 
+            <!-- BEAT LOOP ROLL MODE START (PAD FX1) -->
+            <control>
+                <description>PAD 1 (DECK1) BEAT LOOP ROLL MODE - press - 1/32 Beatloop</description>
+                <group>[Channel1]</group>
+                <key>beatlooproll_0.03125_activate</key>
+                <status>0x97</status>
+                <midino>0x10</midino>
+                <options>
+                    <Normal/>
+                </options>
+            </control>
```

Comment:
You are only mapping Note On controls here, so mixxx will always think these buttons are being held. For the beatlooprolls to work appropriately (non-latching and deactivating on pad release), you also need to map all the note-off messages.

Reply:
I am confused because I have tried it with the controller and it works properly. I don't have the controller with me to double-check what note-off message the controller is sending (I can do that tomorrow), maybe on release it is a note-on message with velocity=0 (seen other controllers do that), or maybe Mixxx is automatically binding the note-off message as well?

Reply:
> or maybe Mixxx is automatically binding the note-off message as well?

Not sure, but I don't think so. 

> I don't have the controller with me to double-check what note-off message the controller is sending

Likely just `0x8X` instead of `0x9X`. Though it might be possible that it still just sends note-on messages with 0 velocity (as you suggested). I haven't seen that on any of my hardware though. It would be great if you could double check with your hardware if you find the time.